mlflow/server/fastapi_security.py

```diff
@@ -0,0 +1,200 @@
+"""
+FastAPI security middleware configuration using native Starlette/FastAPI middleware.
+"""
+
+import logging
+from http import HTTPStatus
+
+from fastapi import FastAPI
+from fastapi.middleware.cors import CORSMiddleware
+from starlette.types import ASGIApp
+
+from mlflow.environment_variables import MLFLOW_ALLOW_INSECURE_CORS, MLFLOW_HOST_HEADER_VALIDATION
+from mlflow.server.security_utils import (
+    CORS_BLOCKED_MSG,
+    HEALTH_ENDPOINTS,
+    INVALID_HOST_MSG,
+    get_allowed_hosts_from_env,
+    get_allowed_origins_from_env,
+    get_default_allowed_hosts,
+    is_api_endpoint,
+    should_block_cors_request,
+    validate_host_header,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+class SecurityHeadersMiddleware:
+    """Middleware to add security headers to all responses."""
+
+    def __init__(self, app: ASGIApp):
+        self.app = app
+
+    async def __call__(self, scope, receive, send):
+        if scope["type"] != "http":
+            return await self.app(scope, receive, send)
+
+        async def send_wrapper(message):
+            if message["type"] == "http.response.start":
+                headers = dict(message.get("headers", []))
+                headers[b"x-content-type-options"] = b"nosniff"
+                headers[b"x-frame-options"] = b"SAMEORIGIN"
+
+                if (
+                    scope["method"] == "OPTIONS"
+                    and message.get("status") == 200
+                    and is_api_endpoint(scope["path"])
+                ):
+                    message["status"] = HTTPStatus.NO_CONTENT
+
+                message["headers"] = list(headers.items())
+            await send(message)
+
+        await self.app(scope, receive, send_wrapper)
+
+
+class HostValidationMiddleware:
+    """Middleware to validate Host headers and prevent DNS rebinding attacks."""
+
+    def __init__(self, app: ASGIApp, allowed_hosts: list[str]):
+        self.app = app
+        self.allowed_hosts = allowed_hosts
+
+    async def __call__(self, scope, receive, send):
+        if scope["type"] != "http":
+            return await self.app(scope, receive, send)
+
+        # Check if this is a health endpoint that bypasses validation
+        path = scope["path"]
+        if path in HEALTH_ENDPOINTS:
+            return await self.app(scope, receive, send)
+
+        # Extract and validate host header
+        headers = dict(scope["headers"])
+        host = headers.get(b"host", b"").decode("utf-8")
```

Comment:
How are we going to enable the requests from MLflow client SDK? Will we manually include the host header?